railties/lib/rails/tasks/statistics.rake

```diff
@@ -1,37 +1,15 @@
 # frozen_string_literal: true
 
-# While global constants are bad, many 3rd party tools depend on this one (e.g
-# rspec-rails & cucumber-rails). So a deprecation warning is needed if we want
-# to remove it.
-STATS_DIRECTORIES ||= [
-  %w(Controllers        app/controllers),
-  %w(Helpers            app/helpers),
-  %w(Jobs               app/jobs),
-  %w(Models             app/models),
-  %w(Mailers            app/mailers),
-  %w(Mailboxes          app/mailboxes),
-  %w(Channels           app/channels),
-  %w(Views              app/views),
-  %w(JavaScripts        app/assets/javascripts),
-  %w(Stylesheets        app/assets/stylesheets),
-  %w(JavaScript         app/javascript),
-  %w(Libraries          lib/),
-  %w(APIs               app/apis),
-  %w(Controller\ tests  test/controllers),
-  %w(Helper\ tests      test/helpers),
-  %w(Job\ tests         test/jobs),
-  %w(Model\ tests       test/models),
-  %w(Mailer\ tests      test/mailers),
-  %w(Mailbox\ tests     test/mailboxes),
-  %w(Channel\ tests     test/channels),
-  %w(Integration\ tests test/integration),
-  %w(System\ tests      test/system),
-]
+STATS_DIRECTORIES = ActiveSupport::Deprecation::DeprecatedObjectProxy.new(
```

Comment:
I wonder if `def self.STATS_DIRECTORIES` wouldn't work better? Unless someone access it with `cons_get`, but that seems a bit far fetched.

Reply:
It seems like all of the major uses are `::STATS_DIRECTORIES` which should still work?
https://github.com/search?q=%3A%3ASTATS_DIRECTORIES&type=code

Reply:
It seems that only works with parentheses:
```ruby
def self.TEST
   "Hi"
end

TEST() # => "Hi"
TEST # `<main>': uninitialized constant TEST (NameError)
::TEST # `<main>': uninitialized constant TEST (NameError)
```

Reply:
Ah good catch, ignore me then.